common/planners/src/plan_expression_chain.rs

```diff
@@ -127,20 +129,25 @@ impl ExpressionChain {
                 self.add_expr(left)?;
                 self.add_expr(right)?;
 
-                let args_type = vec![
-                    left.to_data_type(&self.schema)?,
-                    right.to_data_type(&self.schema)?,
+                let arg_fields = vec![
```

Comment:
CC @zhang2014 , using to_data_field may affect the issue https://github.com/datafuselabs/databend/issues/3572?

Reply:
Looks good to me, it's ok. I will to refactor this after merged pr.